share/completions/gradle.fish

```diff
@@ -1,60 +1,276 @@
-        command gradle -q tasks 2>/dev/null | string match -r '^[[:alnum:]]+ - .*' | string replace ' - ' \t >$gradle_cache_file
+    string trim -- $XDG_CACHE_HOME/gradle-completions/(__fish_md5 -s $argv[1] | string split ' = ')[2]
+end
+
+##############################
+# Configure Tasks Completion #
+##############################
+
+# Outside of a Project
+function __get_gradle_default_task_completion
+    if __contains_gradle_build
+        return
     end
-    cat $gradle_cache_file
+
+    printf '%s\t%s\n' \
+        "buildEnvironment" "Displays all buildscript dependencies declared in root project." \
+        "components" "Displays the components produced by root project." \
+        "dependencies" "Displays all dependencies declared in root project." \
+        "dependencyInsight" "Displays the insight into a specific dependency in root project." \
+        "dependentComponents" "Displays the dependent components of components in root project." \
+        "help" "Displays a help message." \
+        "init" "Initializes a new Gradle build." \
+        "model" "Displays the configuration model of root project." \
+        "projects" "Displays the sub-projects of root project." \
+        "properties" "Displays the properties of root project." \
+        "tasks" "Displays the tasks runnable from root project." \
+        "wrapper" "Generates Gradle wrapper files."
 end
 
-function __contains_gradle_build
-    test -f build.gradle
+complete --command 'gw' --command 'gradle' --command 'gradlew' \
+    --exclusive \
+    --arguments "(__get_gradle_default_task_completion)"
+
+# Inside of a Project
+function __get_gradle_task_completion
+    if not __contains_gradle_build
+        return
+    end
+
+    set -l gradle_cache_file (__create_completion_cache_file "{$PWD}-tasks")
+    if not command test -f $gradle_cache_file -a -s $gradle_cache_file
+        command gradle -q tasks --all 2>/dev/null | string match --regex '^[a-z][A-z:]+.*' | string replace ' - ' \t >$gradle_cache_file
+    end
+
+    # return possible tasks
+    string trim -- <$gradle_cache_file
+end
+
+complete --command 'gw' --command 'gradle' --command 'gradlew' \
+    --condition "__contains_gradle_build" \
+    --exclusive \
+    --arguments "(__get_gradle_task_completion)"
+
+
+###############################
+# Configure Option Completion #
+###############################
+
+function __get_console_completion
+    printf '%s\t%s\n' \
+        "auto" "Use 'rich' in console otherwise 'plain'" \
+        "plain" "Disable color and rich output." \
+        "rich" "Enable color and rich output." \
+        "verbose" "Enable color, rich output, output task names and outcomes at the lifecycle log level"
+end
+
+function __get_property_completion
+    printf '%s\t%s\n' \
+        "org.gradle.cache.reserved.mb" "Reserve Gradle Daemon memory for operations." \
+        "org.gradle.caching" "Set true to enable Gradle build cache." \
```

Comment:
So this is a boolean. Boolean options are typically named so that the thing is done if they are true.

So the "Set true" is just superfluous - "Enable Gradle build cache".